packages/client/src/links/httpBatchLink/streamingHttpUtils.ts

```diff
@@ -11,22 +11,28 @@ import {
 } from '../internals/httpUtils';
 import { HTTPHeaders } from '../types';
 
-/**
- * @param readableStream as given by `(await fetch(url)).body`
- * @param onSingle called for each line of the stream
- * @param parser defaults to `JSON.parse`
- */
-export async function parseJsonStream<TReturn>(
-  readableStream: ReadableStream<Uint8Array> | NodeJS.ReadableStream,
-  onSingle: (index: number, res: TReturn) => void,
-  parser: (text: string) => TReturn = JSON.parse,
-  signal?: AbortSignal,
-): Promise<TReturn | undefined> {
+export async function parseJsonStream<TReturn>(opts: {
+  /**
+   * As given by `(await fetch(url)).body`
+   */
+  readableStream: ReadableStream<Uint8Array> | NodeJS.ReadableStream;
+  /**
+   * Called for each line of the stream
+   */
+  onSingle: (index: number, res: TReturn) => void;
+  /**
+   * Called when the stream is finished
+   */
```

Comment:
You seem to have gotten that comment from an earlier commit where I used multiple callbacks. `parse` is supposed to be basically `JSON.parse` or equivalent and is called on either 1. every line if we have a correct stream, or 2. the entire string if shape isn't recognized.